presto-native-execution/presto_cpp/main/QueryContextManager.cpp

```diff
@@ -96,10 +96,12 @@ std::shared_ptr<core::QueryCtx> QueryContextManager::findOrCreateQueryCtx(
   // is still indexed by the query id.
   static std::atomic_uint64_t poolId{0};
   std::optional<memory::MemoryPool::DebugOptions> poolDbgOpts;
-  const auto debugMemoryPoolNameRegex = queryConfig.debugMemoryPoolNameRegex();
-  if (!debugMemoryPoolNameRegex.empty()) {
+  if (auto debugMemoryPoolNameRegex = queryConfig.debugMemoryPoolNameRegex();
+      !debugMemoryPoolNameRegex.empty()) {
```

Comment:
Shall we use the original way to align with the codebase conventions? It is nice though.

Reply:
Sure. The initial idea was to std::move the value of debugMemoryPoolNameRegex instead of copying it. And I moved the variable into the `if` condition just to limit the scope of the variable being alive. I'll do one step back and only do `std::move` if it works